phases/10-llms-from-scratch/04-pre-training-mini-gpt/code/main.py

```
@@ -0,0 +1,358 @@
+Layer normalization stabilizes training by normalizing activations.
+Position embeddings give the model information about token ordering.
+The causal mask ensures autoregressive generation during training.
+Pre-training on large text corpora teaches the model general language understanding.
+Fine-tuning adapts the pre-trained model to specific downstream tasks."""
+
+    print("Training Mini GPT")
+    print("=" * 65)
+    model = train_mini_gpt(corpus, num_steps=200)
+
+    prompt = list("The transformer".encode("utf-8"))
+    print(f"\nPrompt: 'The transformer'")
```

Ruff (0.15.7)

[error] f-string without any placeholders

Remove extraneous `f` prefix

(F541)